Automattic/jetpack
# New version of the Mailchimp Subscriber Popup Widget

#### Changes proposed in this Pull Request:

* Add new version of the mailchimp popup widget based on the mailchimp block.


#### Testing instructions:






* Make sure that Jetpack is connected to your WordPress.com account.
![Jetpack -> Dashboard](https://user-images.githubusercontent.com/51013565/85169674-cebf3300-b231-11ea-873f-888636b2cb8f.jpg)
* Then go to your WordPress.com account, select the site you are working on, then go to "Marketing" and connect Mailchimp with it:
![](https://user-images.githubusercontent.com/51013565/85187224-b5cf7580-b263-11ea-8dc9-ce5e7bb298ff.jpg)
**Note**: You need to create a Mailchimp account for this, also you need to create a list of subscribers in Mailchimp.
* Go to Jetpack Dashboard > Modules and activate `Extra Sidebar Widgets`
![Dashboard > Modules](https://user-images.githubusercontent.com/51013565/85172529-e0efa000-b236-11ea-87a2-fa17b8756995.jpg)
* Go to the Widgets Page and add the `MailChimp Subscriber Popup (Jetpack)` widget to a sidebar area.
* if you already had a mailchimp code in the widget, then you will see this screen in the widget from:
![](https://user-images.githubusercontent.com/51013565/85173022-e994a600-b237-11ea-8618-df2904aaddfd.jpg)
Since the new form is intented to replace the `code` text area, if you check the checkbox below it then the new form will show (this can not be undone):
![](https://user-images.githubusercontent.com/51013565/85173170-337d8c00-b238-11ea-99e6-699d54c89278.jpg)
* Save the widget
Note: In case you didn't have any code in the mailchimp widget, you will see the new form inmediately.
* Now that the widget is in place, please go to the site frontend and please check the output:
![](https://user-images.githubusercontent.com/51013565/85174032-22357f00-b23a-11ea-9328-10048d93afed.jpg)
* Please try to enter a valid email address in to the form to check that is working well.
* Check the ouput on AMP pages:
![](https://user-images.githubusercontent.com/51013565/85174904-f5826700-b23b-11ea-8d70-a1be170f6dbc.jpg)
* Go back to the widget form, check the check popup mode and save the form:
![](https://user-images.githubusercontent.com/51013565/85175409-fcf64000-b23c-11ea-8459-1ffd524fbbbf.jpg)
* Check that the popup is showing on frontend:
![](https://user-images.githubusercontent.com/51013565/85177857-a0961f00-b242-11ea-846e-70be13217deb.jpg)
* Check the AMP version:
![](https://user-images.githubusercontent.com/51013565/85177927-c8858280-b242-11ea-934b-cfa4ff760581.jpg)

Note: If you close the popup, then it won't appear again after a page refresh. If you want to see the popup again, you would need to delete the local storage and cookies of the site you used to test in the browser. The popup is intended to be shown once.


#### Proposed changelog entry for your changes:

* Update the MailChimp Subscriber Popup Widget.
